sql/src/main/java/io/crate/execution/dml/ShardRequestExecutor.java

```diff
@@ -98,16 +101,21 @@ public ShardRequestExecutor(ClusterService clusterService,
     }
 
     public void execute(RowConsumer consumer, Row parameters, SubQueryResults subQueryResults) {
```

Comment:
Would it make sense to split this class into 2 separate executors one for counting, one to retrieve return values ? Or maybe expose the possibility to handle ActionListener<ShardResponse> to the caller and leave the way the return values are handled to the outside of this class ?